core/src/main/java/hudson/PluginManager.java

```diff
@@ -1666,7 +1666,10 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
 
             JSONArray dependencies = new JSONArray();
             try {
-                Manifest m = new JarFile(t).getManifest();
```

Comment:
Maybe the first try (L1668) could be the try with resources?

Reply:
Since the JarFile is just needed for extracting the Manifest, I thought it would be better to immediatly close the file. But if you prefer to keep it open until the end of the method, I can change that.